docs/pages/guides/keyboard-handling.mdx

```diff
@@ -0,0 +1,262 @@
+
+To listen for keyboard events, use the `Keyboard.addListener` method. This method accepts an event name and a callback function as arguments. The callback function is called with the event data when the keyboard is shown or hidden.
+
+```tsx HomeScreen.tsx
+import { useEffect } from 'react';
+import { Keyboard } from 'react-native';
+
+export default function HomeScreen() {
+  useEffect(() => {
+    const showSubscription = Keyboard.addListener('keyboardDidShow', handleKeyboardShow);
+
+    return () => {
+      showSubscription.remove();
+    };
+  }, []);
+
+  const handleKeyboardShow = event => {
+    // Handle keyboard show event here
+  };
+}
+```
+
+To dismiss the keyboard, use the `dismiss` method.
+
+```tsx DismissKeyboardButton.tsx
+import { Keyboard, Button } from 'react-native';
+
+export default function DismissKeyboardButton() {
+  return <Button title="Dismiss keyboard" onPress={Keyboard.dismiss} />;
+}
+```
+
+### Keyboard avoiding view
+
+The `KeyboardAvoidingView` is a component that automatically adjusts its height, position, or bottom padding based on the keyboard height to remain visible while the keyboard is displayed.
+
+```tsx HomeScreen.tsx
+import { KeyboardAvoidingView, TextInput } from 'react-native';
+
+export default function HomeScreen() {
+  return (
+    <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : undefined} style={{ flex: 1 }}>
+      <TextInput placeholder="Type here..." />
+    </KeyboardAvoidingView>;
+  );
+}
+```
+
+In the above example, the height of the `KeyboardAvoidingView` automatically adjusts based on the device's keyboard height, which ensures that the input is always visible.
+
+When using a Bottom Tab navigator on Android, you might notice that focusing on an input field causes the bottom tabs to be pushed above the keyboard. To address this issue, add the `softwareKeyboardLayoutMode` property to your Android configuration in [app confg](/workflow/configuration/) and set it to `pan`.
+
+```json app.json
+"expo" {
+  "android": {
+    "softwareKeyboardLayoutMode": "pan"
+  }
+}
+```
+
+After adding this property, restart your app so that the changes take effect.
+
+## Advanced keyboard handling
+
+For more complex keyboard interactions, consider using the [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller) library. It offers additional functionality beyond the built-in React Native keyboard APIs, providing consistency across Android and iOS with minimal configuration and offering the native feel users expect.
+
+### Install keyboard controller
+
+To get started with [Keyboard Controller](https://kirillzyusko.github.io/react-native-keyboard-controller), install it in your Expo project:
+
+<Terminal cmd={['$ npx expo install react-native-keyboard-controller']} />
+
+Keyboard Controller uses native code, so you need to prebuild your app:
+
+<Terminal cmd={['$ npx expo prebuild --clean']} />
```

Comment:
You would only need to explicitly run this if you were building locally with `npx expo run`. I would instead say that you will need to use a development build since this package is not in Expo Go. (then link to development build)

Reply:
Adding to what Keith mentioned in the above comment, I think we can remove the prebuild command from here and instead:

- Add a subsection called "prerequisites" and mention that this guide assumes folks are creating development builds. Something like the following:

```md
The following steps described use a [development build](/develop/development-builds/introduction/) since this library is not included in Expo Go. See [create a development build](/develop/development-builds/create-a-build/) for more information.
```